### PR TITLE
Auto aux update check

### PR DESCRIPTION
**Pull request by shahronak47**
Clickup task - https://app.clickup.com/t/868763th2 

I think I have a fix, but let's test it when we have aux data to update before merging.

### CODE REVIEW

File: R/auto_aux_update.R [hunk 1]
```diff
@@ -99,31 +99,38 @@ auto_aux_update <- function(measure = NULL,
...
-                )
-
-  cli::cli_h2("File updated status.")
   out <- aux_file_last_updated(maindir, names(dependencies), branch)
+  if(length(aux_fns) > 0) {
+    # Check if the file has updated, only then update the metadata file
+    # If the files that we wanted to updated were updated today only then write the file
+    temp_dat <- out |>
+        dplyr::filter(filename %in% paste0(aux_fns, ".qs")) |>
+        dplyr::filter(as.Date(time_last_update) == Sys.Date())
```
Comment (randrescastaneda) [inline R/auto_aux_update.R]: I don't Think this will work well, especially since it is possible to update the same file several times a day

**Review by randrescastaneda — CHANGES_REQUESTED** [review]
Hi @shahronak47 , 

I noticed you're relying on the modified date to track changes in aux files. While that works, it could be risky if they're edited multiple times in a day.

For enhanced security and clarity, I propose using the hashes created by the `pipfun::pip_sign_save()` function. This function calculates and saves a unique hash each time an aux file is updated, ensuring precise identification of different versions.

Please review the details of `pipfun::pip_sign_save()` and the hash files saved. Let me know if you agree with a hash-based approach. If not, we can explore further improvements to the current date-based method.

Thanks.

**Comment by shahronak47** [comment]
> Hi @shahronak47 ,
> 
> I noticed you're relying on the modified date to track changes in aux files. While that works, it could be risky if they're edited multiple times in a day.
> 
> For enhanced security and clarity, I propose using the hashes created by the `pipfun::pip_sign_save()` function. This function calculates and saves a unique hash each time an aux file is updated, ensuring precise identification of different versions.
> 
> Please review the details of `pipfun::pip_sign_save()` and the hash files saved. Let me know if you agree with a hash-based approach. If not, we can explore further improvements to the current date-based method.
> 
> Thanks.

Instead of today's date we can look if the file is updated in last x mins. We can define x as 5 or 10 mins. Using hash is more safer  however, it can be a lot of work to get that hash for each file which is updated because `pip_sign_save` is not explicitly called in our function.

**Comment by randrescastaneda** [comment]
> > Hi @shahronak47 ,
> > I noticed you're relying on the modified date to track changes in aux files. While that works, it could be risky if they're edited multiple times in a day.
> > For enhanced security and clarity, I propose using the hashes created by the `pipfun::pip_sign_save()` function. This function calculates and saves a unique hash each time an aux file is updated, ensuring precise identification of different versions.
> > Please review the details of `pipfun::pip_sign_save()` and the hash files saved. Let me know if you agree with a hash-based approach. If not, we can explore further improvements to the current date-based method.
> > Thanks.
> 
> Instead of today's date we can look if the file is updated in last x mins. We can define x as 5 or 10 mins. Using hash is more safer however, it can be a lot of work to get that hash for each file which is updated because `pip_sign_save` is not explicitly called in our function.

Hi @shahronak47 , 

Right where the aux file is saved, there is another file called `yyy_datasignature`, where `yyy` stands for the name of the auxiliary file (e.g., cpi, gdp, ppp, etc). That file contains the hash. Just read it before the `*update()` process starts. Does it make sense? Please, let me know. Best

**Pull request merged**